theano/tensor/tests/test_subtensor.py

```diff
@@ -362,6 +362,51 @@ def test_ellipsis(self):
             assert_equal(tval.shape, numpy_tval.shape)
             assert_array_equal(tval, numpy_tval)
 
+    def test_boolean(self):
+        numpy_n = np.arange(6, dtype=self.dtype).reshape((2, 3))
+        n = self.shared(numpy_n)
+
+        # indexing with a comparison (should translate to a boolean mask)
+        assert_array_equal(numpy_n[numpy_n > 2], n[n > 2].eval())
+        assert_array_equal(numpy_n[[0], numpy_n[0] > 2], n[[0], n[0] > 2].eval())
+        assert_array_equal(numpy_n[[1], numpy_n[0] > 2], n[[1], n[0] > 2].eval())
+
+        # indexing with a mask for some dimensions
+        mask = np.array([True, False])
+        assert_array_equal(numpy_n[mask], n[mask].eval())
+
+        # indexing with a mask for the second dimension
+        mask = np.array([True, False, True])
+        assert_array_equal(numpy_n[0, mask], n[0, mask].eval())
+        assert_array_equal(numpy_n[:, mask], n[:, mask].eval())
+        assert_array_equal(numpy_n[:, mask], n[:, self.shared(mask)].eval())
+
+        # indexing with a boolean array
+        mask = np.array([[True, False, True], [False, False, True]])
+        assert_array_equal(numpy_n[mask], n[mask].eval())
+        assert_array_equal(numpy_n[mask], n[self.shared(mask)].eval())
+
+        # for the GpuArray tests, self.shared will create GpuArray variables,
+        # but we also need to test the combination of GPU and normal TensorVariables
+        n_cpu = theano.shared(numpy_n)
+        assert_array_equal(numpy_n[mask], n[theano.shared(mask)].eval())
```

Comment:
this won't test on the GPU. The GPU tests don't enable the GPU by default. They need to manually pass the mode. You can use self.mode as the mode to use to enable the GPU optimization to make the test do what you want.

But I don't think we need to test that. I would just remove that test. This is the normal workflow and should work without problems.